### PR TITLE
Add Path Iterator to return results like WalkPath would visit

### PR DESCRIPTION
**Pull request by mkeeler**
## Description

I want to avoid the callbacks of WalkPath and be able to iterate over a snapshot of the tree so I made a PathIterator to do just that. It iterates through the tree identically to what WalkPath would except now the iteration is split across multiple invocations of Next.

## TODOS
- [x] Add some PathIterator unit tests as a duplicate of the WalkPath unit tests.

### CODE REVIEW

File: path_iter.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// the Node.WalkPath method will.
+type PathIterator[T any] struct {
+	node *Node[T]
+	path []byte
+	done bool
+}
+
+// Next returns the next node in order
+func (i *PathIterator[T]) Next() ([]byte, T, bool) {
+	// This is mostly just an asyncrhonous implementation of the WalkPath
```
Comment (kisunji) [inline path_iter.go:16]:
```suggestion
	// This is mostly just an asynchronous implementation of the WalkPath
```

File: path_iter.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package iradix
+
+import "bytes"
+
+// PathIterator is used to iterate over a set of nodes from the root
+// down to a specified path. This will iterate overthe same values that
```
Comment (kisunji) [inline path_iter.go:6]:
```suggestion
// down to a specified path. This will iterate over the same values that
```
